src/class-wp-import.php

```diff
@@ -0,0 +1,1420 @@
+	 * or falls back to the current user in case of error with either of the previous.
+	 */
+	public function get_author_mapping() {
+		if ( ! isset( $_POST['imported_authors'] ) ) {
+			return;
+		}
+
+		$create_users = $this->allow_create_users();
+
+		foreach ( (array) $_POST['imported_authors'] as $i => $old_login ) {
+			// Multisite adds strtolower to sanitize_user. Need to sanitize here to stop breakage in process_posts.
+			$santized_old_login = sanitize_user( $old_login, true );
+			$old_id             = isset( $this->authors[ $old_login ]['author_id'] ) ? intval( $this->authors[ $old_login ]['author_id'] ) : false;
+
+			if ( ! empty( $_POST['user_map'][ $i ] ) ) {
+				$user = get_userdata( (int) $_POST[ 'user_map' ][ $i ] );
+				if ( isset( $user->ID ) ) {
+					if ( $old_id ) {
+						$this->processed_authors[ $old_id ] = $user->ID;
+					}
+					$this->author_mapping[ $santized_old_login ] = $user->ID;
+				}
+			} elseif ( $create_users ) {
+				if ( ! empty( $_POST['user_new'][ $i ] ) ) {
+					$user_id = wp_create_user( $_POST['user_new'][ $i ], wp_generate_password() );
+				} elseif ( '1.0' !== $this->version ) {
+					$user_data = array(
+						'user_login'   => $old_login,
+						'user_pass'    => wp_generate_password(),
+						'user_email'   => isset( $this->authors[ $old_login ]['author_email'] ) ? $this->authors[ $old_login ]['author_email'] : '',
+						'display_name' => $this->authors[ $old_login ]['author_display_name'],
+						'first_name'   => isset( $this->authors[ $old_login ]['author_first_name'] ) ? $this->authors[ $old_login ]['author_first_name'] : '',
+						'last_name'    => isset( $this->authors[ $old_login ]['author_last_name'] ) ? $this->authors[ $old_login ]['author_last_name'] : '',
+					);
+					$user_id   = wp_insert_user( $user_data );
+				}
+
+				if ( ! is_wp_error( $user_id ) ) {
+					if ( $old_id ) {
+						$this->processed_authors[ $old_id ] = $user_id;
+					}
+					$this->author_mapping[ $santized_old_login ] = $user_id;
+				} else {
+					printf( __( 'Failed to create new user for %s. Their posts will be attributed to the current user.', 'wordpress-importer' ), esc_html( $this->authors[ $old_login ]['author_display_name'] ) );
+					if ( defined( 'IMPORT_DEBUG' ) && IMPORT_DEBUG ) {
+						echo ' ' . $user_id->get_error_message();
+					}
+					echo '<br />';
+				}
+			}
+
+			// Failsafe: if the user_id was invalid, default to the current user.
+			if ( ! isset( $this->author_mapping[ $santized_old_login ] ) ) {
+				if ( $old_id ) {
+					$this->processed_authors[ $old_id ] = get_current_user_id();
+				}
+				$this->author_mapping[ $santized_old_login ] = get_current_user_id();
+			}
+		}
+	}
+
+	/**
+	 * Create new categories based on import information.
+	 *
+	 * Doesn't create a new category if its slug already exists.
+	 */
+	public function process_categories() {
+		$this->categories = (array) apply_filters( 'wp_import_categories', $this->categories );
```

Comment:
Looks like the `(array)` is a new introduction.

* Should this be in a CS PR ?
* Should the array cast be done before or after the `empty()` check ? or should there maybe not be an array cast at all and the `empty()` check extended with an `is_array()` check ?